pkgs/tools/package-management/rpm/darwin.patch

```diff
@@ -0,0 +1,33 @@
+--- src/lib/backend/ndb/rpmidx.c	2020-12-14 04:42:43.000000000 -0800
++++ src/lib/backend/ndb/rpmidx.c	2020-12-22 08:02:51.000000000 -0800
+@@ -16,7 +16,14 @@
+ #include <sys/mman.h>
+ #include <errno.h>
+ 
++#ifdef __APPLE__
++#include <machine/endian.h>
++#include <libkern/OSByteOrder.h>
++#define htole32(x) OSSwapHostToLittleInt32(x)
++#define le32toh(x) OSSwapLittleToHostInt32(x)
++#else
+ #include <endian.h>
++#endif /* __APPLE__ */
```

Comment:
cc @pmatilai Is this something upstream is interested in?

Reply:
https://github.com/rpm-software-management/rpm/pull/1465 :)

Reply:
Ok. Please also add the link to the patch file as suggested below.

Reply:
Done